### PR TITLE
MNT: Compat with pytest>=8

### LINKED ISSUE #14: Not compatible with pytest 8.0 ?
I am not sure what is going on because I cannot reproduce this locally with pytest 8.0rc but I see this in astropy CI (https://github.com/astropy/astropy/issues/15810):

```
File "_pytest/config/__init__.py", line 1309, in _preparse
    self.pluginmanager.load_setuptools_entrypoints("pytest11")
  File "pluggy/_manager.py", line 399, in load_setuptools_entrypoints
    self.register(plugin, name=ep.name)
  File _pytest/config/__init__.py", line 496, in register
    ret: Optional[str] = super().register(plugin, name)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "pluggy/_manager.py", line 163, in register
    self._verify_hook(hook, hookimpl)
  File "pluggy/_manager.py", line 328, in _verify_hook
    raise PluginValidationError(
pluggy._manager.PluginValidationError: Plugin 'pytest_filter_subpackage' for hook 'pytest_ignore_collect'
hookimpl definition: pytest_ignore_collect(path, config)
Argument(s) {'path'} are declared in the hookimpl but can not be found in the hookspec
```

Does `Argument(s) {'path'} are declared in the hookimpl but can not be found in the hookspec` mean anything to you?

### PR DESCRIPTION
**Pull request by pllim**
Fix #14 as suggested in https://github.com/pytest-dev/pytest/issues/11779#issuecomment-1879387034

~Out of scope: Fix the CI (looks very outdated but I don't have time to deal with it)~

### CODE REVIEW

**Comment by pllim** [comment]
Not sure how to easily add a pytest-dev job to this Azure stuff.

File: pytest_filter_subpackage/plugin.py
Comment (neutrinoceros) [inline pytest_filter_subpackage/plugin.py:11]: FYI this boilerplate isn't strictly necessary for pytest: you may just use `pytest.version_tuple`. Note however that it was only introduced in pytest 7.0 so some tinkering is still needed for maximum portability.
```
PYTEST_GE_8 = getattr(pytest, "version_tuple", (0, 0, 0)) >= (8, 0, 0)
```

Comment (pllim) [reply pytest_filter_subpackage/plugin.py:11]: These plugins generally support really old pytest versions until there is no choice, so I am not sure if using `version_tuple` is practical here.  Thanks for the suggestion anyway!

Comment (neutrinoceros) [reply pytest_filter_subpackage/plugin.py:11]: Actually my suggestion should work with any version. I also note that your solution involves `packaging`, which isn't part of the standard library (albeit very common, *and* currently a dependency to pytest), so you'd at least need to declare it as a dependency in the package metadata.

**Comment by bsipocz** [comment]
> Not sure how to easily add a pytest-dev job to this Azure stuff.

Can we just get rid of altogether it and copy-paste the GHA config over from doctestplus/remote-data?

File: pytest_filter_subpackage/plugin.py
Comment (pllim) [reply pytest_filter_subpackage/plugin.py:11]: @bsipocz , do you have a preference on version check? I copied this from pytest-doctestplus . Maybe we should decide and then make both consistent?

Comment (bsipocz) [reply pytest_filter_subpackage/plugin.py:11]: I'm not a fan of fixing what's not broken, so definitely prefer to keep using `Version` as we do that everywhere else, including astropy core.

**Comment by pllim** [comment]
@bsipocz , good idea on the copy paste. I copied from doctestplus instead since I think that one was updated more recently? Anyway, shouldn't matter if I copied from doctestplus or remote-data.

However, now that we have a proper pytest-dev job, it is failing because we have not merged this PR because apparently doctestplus is a test dependency here:

* https://github.com/scientific-python/pytest-doctestplus/pull/236

**Comment by bsipocz** [comment]
> However, now that we have a proper pytest-dev job, it is failing because we have not merged this PR because apparently doctestplus is a test dependency here:

I've restarted CI after the upstream merge.

File: .github/workflows/python-tests.yml
Comment (bsipocz) [inline .github/workflows/python-tests.yml:61]: If all else is looking good, I'll add the 8.0.x jobs here, too

File: setup.cfg
Comment (bsipocz) [inline setup.cfg:7]: 💯

**Review by bsipocz — APPROVED** [review]
Looks all good. I suppose the repeated ~20-25 lines could be dumped into a shared functions, but then it's borderline than just simply keep the repeats.

However, I think we need to add tests with the 8.0.x branch. I'm adding a commit with that, and if all CI is green, will go ahead and merge.

File: CHANGES.rst
Comment (bsipocz) [inline CHANGES.rst:1]: Given the stable metadata below, I would consider bumping the version number to either 0.2 or even 1.0

**Comment by bsipocz** [comment]
Thanks @pllim!

**Comment by pllim** [comment]
Thanks for cleaning this up and merging!

File: pytest_filter_subpackage/plugin.py
Comment (pllim) [reply pytest_filter_subpackage/plugin.py:11]: Thanks anyway, @neutrinoceros . We can always revisit the versioning stuff in a separate PR.

**Comment by pllim** [comment]
p.s. I pushed a commit directly to main to bump change log section to 0.2 to be consistent with milestone. FYI.

**Comment by bsipocz** [comment]
yeap, I just added the milestone that was already around, but as I said above, I would not be against calling it v1.0 in the next release.

**Comment by pllim** [comment]
Yeah, maybe it is time for 1.0.

But... if you have plans to move this to Scientific Python, maybe we should wait for that first? Or is this too astropy specific?

**Comment by bsipocz** [comment]
I don't see straight away who would be interested in using this (the main benefit is that this collects both from the source and docs, but if the narrative docs is not structured in the same submodule way, or doctesting is not done then there is no point using this.)

So, we can certainly upstream it if there are other users, but first would need to find some interest.

**Pull request merged**
